contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,354 @@
+
+        /// pool deposit
+        IERC20(USDC).approve(maplePool, amount);
+        IMaplePool(maplePool).deposit(amount);
+
+        /// stake pool FDT for MPL rewards
+        uint256 scaledDepositAmount = amount * SCALING_FACTOR;
+        IMaplePool(maplePool).increaseCustodyAllowance(
+            mapleRewards,
+            scaledDepositAmount
+        );
+        IMapleRewards(mapleRewards).stake(scaledDepositAmount);
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+
+        if (pcvOracle != address(0)) {
+            IPCVOracle(pcvOracle).updateIlliquidBalance(
+                endingRecordedBalance - startingRecordedBalance
+            );
+        }
+
+        emit Deposit(msg.sender, amount);
+    }
+
+    /// @notice function to start the cooldown process to withdraw
+    /// 1. lp lockup on deposit --> 90 days locked up and can't withdraw
+    /// 2. cool down period, call intend to withdraw -->
+    ///   must wait 10 days before withdraw after calling intend to withdraw function
+    /// 3. after cool down and past the lockup period,
+    ///    have 2 days to withdraw before cool down period restarts.
+    function signalIntentToWithdraw() external onlyPCVController {
+        IMaplePool(maplePool).intendToWithdraw();
+    }
+
+    /// @notice function to cancel a withdraw
+    /// should only be used to allow a transfer when doing a withdrawERC20 call
+    function cancelWithdraw() external onlyPCVController {
+        IMaplePool(maplePool).cancelWithdraw();
+    }
+
+    /// @notice function that emits an event tracking profits and losses
+    /// since the last contract interaction
+    /// then writes the current amount of PCV tracked in this contract
+    /// to lastRecordedBalance
+    /// @return the amount deposited after adding accrued interest or realizing losses
+    function accrue() external nonReentrant whenNotPaused returns (uint256) {
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        _recordPNL(); /// update deposit amount and fire harvest event
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+
+        if (pcvOracle != address(0)) {
+            /// if any amount of PCV is withdrawn and no gains, delta is negative
+            IPCVOracle(pcvOracle).updateIlliquidBalance(
+                endingRecordedBalance - startingRecordedBalance
+            );
+        }
+
+        return lastRecordedBalance; /// return updated pcv amount
+    }
+
+    /// @notice permissionless function to harvest rewards before withdraw
+    function harvest() public nonReentrant whenNotPaused {
+        uint256 preHarvestBalance = IERC20(mapleToken).balanceOf(address(this));
+
+        IMapleRewards(mapleRewards).getReward();
+
+        uint256 postHarvestBalance = IERC20(mapleToken).balanceOf(
+            address(this)
+        );
+
+        emit Harvest(
+            mapleToken,
+            int256(postHarvestBalance - preHarvestBalance),
```

Comment:
add comment here that this should never be negative, the smallest number it can be is 0, therefore no need to use SafeCast on cast